resolvers.js

```diff
@@ -203,6 +295,39 @@ const resolvers = {
         }
       );
     },
+
+    updateNoteCategory: async (parent, { noteId, categoryId, index }, context) => {
+      assertAuthenticated(context);
+      await verifyNoteBelongsToUser(context, noteId);
+
+      const noteDocument = await Note.findOne({ _id: noteId });
+      const currentCategoryDocument = await Category.findOne({ _id: noteDocument.categoryId });
+      arrayRemoveItem(currentCategoryDocument.notes, noteDocument._id);
+      await currentCategoryDocument.save();
+
+      const newCategoryDocument = await Category.findOne({ _id: categoryId });
+
+      const databaseDocument = await Database.findOne({ _id: newCategoryDocument.databaseId });
+
+      // check currentview, to determine whether index will be used
+      databaseDocument.currentView === 'board'
+        ? newCategoryDocument.notes.splice(index, 0, noteDocument._id)
+        : newCategoryDocument.notes.push(noteDocument._id);
+      await newCategoryDocument.save();
+
+      await Note.findOneAndUpdate(
+        { _id: noteId },
+        { categoryId: categoryId },
+        {
+          new: true,
+          userFindAndModify: false
+        }
+      );
+
+      await databaseDocument.save();
+      return databaseDocument;
+    },
```

Comment:
there's probably a better way to write this without so many awaits. this function call is probably very slow as we are waiting for network requests to resolve at every line.

can look at batch updating